wa/commands/show.py

```diff
@@ -0,0 +1,93 @@
+#    Copyright 2014-2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from subprocess import call, Popen, PIPE
+
+from wa import Command
+from wa.framework import pluginloader
+from wa.framework.exception import NotFoundError
+from wa.framework.target.descriptor import get_target_descriptions
+from wa.utils.doc import (strip_inlined_text, get_rst_from_extension,
+                          get_params_rst, underline)
+from wa.utils.misc import which
+
+from devlib.utils.misc import escape_double_quotes
+
+class ShowCommand(Command):
+
+    name = 'show'
+    description = 'Display documentation for the specified plugin (workload, instrument, etc.).'
+
+    def initialize(self, context):
+        self.parser.add_argument('plugin', metavar='PLUGIN',
+                                 help='The name of the plugin to display documentation for.')
+
+    def execute(self, state, args):
+        name = args.plugin
+        rst_output = None
+
+        plugin = get_plugin(name)
+        if plugin:
+            rst_output = get_rst_from_extension(plugin)
+        else:
+            target = get_target(name)
+            if target:
+                rst_output = get_rst_from_target(target)
+
+        if not rst_output:
+            raise NotFoundError('Could not find plugin or alias "{}"'.format(name))
+
+        if which('pandoc'):
+            p = Popen(['pandoc', '-f', 'rst', '-t', 'man'], stdin=PIPE, stdout=PIPE, stderr=PIPE)
+            output, _ = p.communicate(rst_output)
+            # Make sure to double escape back slashes
+            output = output.replace('\\', '\\\\\\')
+            # Correctly format the tile of the man page
+            output = output.replace('.SH', '.TH', 1)
+            call('echo "{}" | man -l -'.format(escape_double_quotes(output)), shell=True)
+        else:
+            print rst_output
+
+def get_plugin(name):
+    for kind in pluginloader.kinds:
+        results = pluginloader.list_plugins(kind)
+        for result in results:
+            if name == result.name:
+                return result
+            if hasattr(result, 'alias'):
+                for alias in result.alias:
+                    if name == alias:
+                        return result
+
+def get_target(name):
```

Comment:
For the sake of clarity, I would rename this to something like `get_target_description`